### PR TITLE
Refactor: move sanity check out of while loop for optimizing gas usage

### PR DESCRIPTION
**Pull request by ujenjt**
Move sanity check out of while loop for optimizing gas usage

photo_2023-02-12 20 26 47

### CODE REVIEW

**Review by TheDZhon — APPROVED** [review]
LGTM

**Review by arwer13 — APPROVED** [review]
👍

File: contracts/0.8.9/oracle/AccountingOracle.sol [hunk 1]
```diff
@@ -683,6 +683,8 @@ contract AccountingOracle is BaseOracle {
 
     function _processExtraDataItems(bytes calldata data, ExtraDataIterState memory iter) internal {
         uint256 dataOffset = iter.dataOffset;
+        uint256 maxNodeOperatorsPerItem = 0;
+        uint256 maxNodeOperatorItemIndex = 0;
 
         /// @solidity memory-safe-assembly
```
Comment (folkyatina) [inline contracts/0.8.9/oracle/AccountingOracle.sol:689]: Not related to the changes, but I believe that it should be placed right above the `assembly {}` block according to the docs https://docs.soliditylang.org/en/latest/assembly.html#memory-safety

**Review by folkyatina — APPROVED** [review]
Looks fine. LGTM

File: contracts/0.8.9/oracle/AccountingOracle.sol (hunk 1 above)
Comment (skozin) [reply contracts/0.8.9/oracle/AccountingOracle.sol:689]: I cannot see anything in the docs about memory safety that would imply this

Comment (skozin) [reply contracts/0.8.9/oracle/AccountingOracle.sol:689]: could you please point to a specific place?

File: contracts/0.8.9/oracle/AccountingOracle.sol [hunk 2]
```diff
@@ -714,17 +716,27 @@ contract AccountingOracle is BaseOracle {
...
         }
+
+        if (maxNodeOperatorsPerItem > 0) {
+            IOracleReportSanityChecker(LOCATOR.oracleReportSanityChecker())
+                .checkNodeOperatorsPerExtraDataItemCount(maxNodeOperatorItemIndex, maxNodeOperatorsPerItem);
+        }
     }
 
-    function _processExtraDataItem(bytes calldata data, ExtraDataIterState memory iter) internal {
+    function _processExtraDataItem(bytes calldata data, ExtraDataIterState memory iter) internal returns (uint256 nodeOpsProcessed) {
```
Comment (skozin) [inline contracts/0.8.9/oracle/AccountingOracle.sol:739]: I'd propose changing this to `nodeOpsCount` and either removing the `nodeOpsCount` variable below or removing the name here

File: contracts/0.8.9/oracle/AccountingOracle.sol [hunk 3]
```diff
@@ -714,17 +716,27 @@ contract AccountingOracle is BaseOracle {
             if (itemType == EXTRA_DATA_TYPE_EXITED_VALIDATORS ||
                 itemType == EXTRA_DATA_TYPE_STUCK_VALIDATORS
             ) {
-                _processExtraDataItem(data, iter);
+                uint256 nodeOpsProcessed = _processExtraDataItem(data, iter);
```
Comment (skozin) [inline contracts/0.8.9/oracle/AccountingOracle.sol:719]: see the comment about variable name below

File: contracts/0.8.9/oracle/AccountingOracle.sol (hunk 1 above)
Comment (skozin) [reply contracts/0.8.9/oracle/AccountingOracle.sol:689]: ahhhh, I missed that this comment is in wrong place. I thought you're talking about the two `uint256` variables above

Comment (skozin) [reply contracts/0.8.9/oracle/AccountingOracle.sol:689]: sorry

**Review by skozin — APPROVED** [review]

**Pull request merged**